### DISCUSSION TITLE
Submit form from a parent

### DISCUSSION THREAD

**Discussion by vocko** [react-hook-form/react-hook-form#4923] [Q&A] [answered]
Is there a way to submit the form from the parent component? In example if I have the form in a modal and the submit button would be part of the modal (= out of the form).

I don't want to be moving the form to the modal as it doesn't make sense conceptually, would love to find a way to submit from the outside, any ideas?

**Very simplified example:**
```
const FormComponent: React.FC<IFormComponentProps> = (props: IFormComponentProps) => {
    
    const form = useForm();
    const { handleSubmit } = form;
    
    return <form onSubmit={handleSubmit(onSubmit)>
        // form components here
    </form>;
};

const ParentComponent: ReactFC = () => {
    return <div>
        <FormComponent onSubmit={(data) => console.debug(data)} />
        <button onClick={submitForm()} />
    </div>
};
```

Does it make sense what I'm trying to do?

**Comment by bluebill1049** [comment]
move `useForm` to your parent, and invoke `handleSubmit()()` at the parent.

**Reply by vocko** [reply]
But that makes the FormComponent not reusable and also placing some unrelated logic to the parent. In my example the Parent will be a Tab or Modal that can hold various context and it is a poor architecture design to hold that state there. Why would modal have useForm when most of the times it is used, it's content doesn't have any form?

The only thing I want it the button on the outer component (Modal, Tab) be able to submit the form.
I was trying to use form ref and do `formRef.current.submit()`, however, it triggers the default form action and the handler doesn't seem to be attached to it.

**Comment by leapful** [comment, marked as answer]
@vocko @bluebill1049 

You can use the built-in attribute "form" of button element instead of using "ref".

For example:

```

const Form = ({ id }) => {
   const { handleSubmit, register } = useForm();

   // ...Handle onSubmit

   return (
               <form id={id} onSubmit={handleSubmit(onSubmit)}>
                    <input {...register('test')}/>
               </form>
    );
}

const Parent = () => {

   return <>
                  <Form id="test">
                  <button form="test">Submit</button>
              </>
}

```

**Reply by vocko** [reply]
Thanks @leapful this actually works. I did not even know such attribute exists!

**Comment by simplenotezy** [comment]
Although all of these solutions work, they feel hacky and like workarounds. Would love to see an integrated way to submit forms from outside the component.
